leveldown.d.ts

```diff
@@ -1,79 +1,74 @@
-    createIfMissing?: boolean;
-    errorIfExists?: boolean;
-    compression?: boolean;
-    cacheSize?: number;
-    writeBufferSize?: number;
-    blockSize?: number;
-    maxOpenFiles?: number;
-    blockRestartInterval?: number;
-    maxFileSize?: number;
-  }
+export interface LevelDown
+  extends LevelDownConstructor, AbstractLevelDOWN<
+  BaseType, BaseType,
+  LevelDownOptions,
+  LevelDownPutOptions,
+  LevelDownGetOptions,
+  LevelDownDeleteOptions,
+  LevelDownIteratorOptions,
+  LevelDownBatchOptions> {
 
-  export interface LevelDownPutOptions {
-    sync?: boolean
-  }
+  approximateSize(start: BaseType, end: BaseType, cb: (err?: any) => void): void;
+  compactRange(start: BaseType, end: BaseType, cb: (err?: any) => void): void;
+  getProperty(property: string): string;
+  destroy(location: string, cb: (err?: any) => void): void;
+  repair(location: string, cb: (err?: any) => void): void;
 
-  export interface LevelDownGetOptions {
-    fillCache?: boolean;
-    asBuffer?: boolean;
-  }
+  iterator(options?: LevelDownIteratorOptions & AbstractIteratorOptions<BaseType>): LevelDownIterator
+}
+
+interface LevelDownConstructor {
+  new (location: string): LevelDown
+  (location: string): LevelDown
+}
 
-  export interface LevelDownDeleteOptions {
-    sync?: boolean;
-  }
+export interface LevelDownOptions {
+  createIfMissing?: boolean;
+  errorIfExists?: boolean;
+  compression?: boolean;
+  cacheSize?: number;
+  writeBufferSize?: number;
+  blockSize?: number;
+  maxOpenFiles?: number;
+  blockRestartInterval?: number;
+  maxFileSize?: number;
+}
 
-  export interface LevelDownIteratorOptions<K, V> {
-    gt?: K;
-    gte?: K;
-    lt?: K;
-    lte?: K;
-    reverse?: boolean;
-    keys?: boolean;
-    values?: boolean;
-    limit?: number;
-    fillCache?: boolean;
-    keyAsBuffer?: boolean;
-    valueAsBuffer?: boolean;
-  }
+export interface LevelDownPutOptions {
+  sync?: boolean
+}
+
+export interface LevelDownGetOptions {
+  fillCache?: boolean;
+  asBuffer?: boolean;
+}
 
-  export interface LevelDownBatchOptions {
-    sync?: boolean;
-  }
+export interface LevelDownDeleteOptions {
+  sync?: boolean;
+}
+
+export interface LevelDownIteratorOptions {
+  fillCache?: boolean;
+
+  keyAsBuffer?: boolean;
+  valueAsBuffer?: boolean;
+}
+
+export interface LevelDownBatchOptions {
+  sync?: boolean;
+}
 
-  export interface LevelDownIterator<TKey> extends Abstract.Iterator {
-    seek(key: TKey);
-    binding: any;
-    cache: any;
-    finished: any;
-    fastFuture: any;
-  }
+export interface LevelDownIterator extends AbstractIterator<BaseType, BaseType> {
+  seek(key: BaseType);
+  binding: any;
+  cache: any;
+  finished: any;
+  fastFuture: any;
 }
-declare function leveldown<
-  TKey=any,
-  TValue=any
-  >(location: string)
-  : leveldown.LevelDown<TKey, TValue>;
 
-export = leveldown;
\ No newline at end of file
+declare const LevelDown: LevelDownConstructor
+export default LevelDown;
```

Comment:
I'd like to suggest also `export { LevelDown }`, the reason is as same as the other reviews.

Reply:
sure, I'll update the pr's in awhile.

Reply:
@zixia why do you want `export { LevelDown }` too? 

the `LevelDown` interface is already exported further up and can be imported with 

```ts
import { LevelDown } from './leveldown';
```